client/src/components/EditProfile/EditPhoto/EditPhoto.tsx

```diff
@@ -1,77 +1,93 @@
-import { useState } from 'react';
-import Paper from '@material-ui/core/Paper';
-import Box from '@material-ui/core/Box';
-import Grid from '@material-ui/core/Grid';
-import Button from '@material-ui/core/Button';
-import Avatar from '@material-ui/core/Avatar';
-import Typography from '@material-ui/core/Typography';
+import { Box, TextField, Grid, Button, IconButton, Avatar, Typography } from '@material-ui/core';
 import useStyles from './useStyles';
 import DeleteSharpIcon from '@material-ui/icons/DeleteSharp';
-import IconButton from '@material-ui/core/IconButton';
+import { uploadImage } from '../../../helpers/APICalls/uploadImage';
+import editProfile from '../../../helpers/APICalls/editProfile';
+import { useAuth } from '../../../context/useAuthContext';
+import { useSnackBar } from '../../../context/useSnackbarContext';
 
 export default function EditPhoto(): JSX.Element {
-  const [file, setFile] = useState<string>('');
+  const { loggedInUserProfile, fetchProfileAndUpdateContext } = useAuth();
+  const { updateSnackBarMessage } = useSnackBar();
 
   const classes = useStyles();
 
-  function importData() {
-    const input: HTMLInputElement = document.createElement('input');
-    input.type = 'file';
-    input.accept = 'image/png, image/jpeg';
-    input.onchange = () => {
-      const files = input.files;
-      setFile(files === null ? '' : URL.createObjectURL(files[0]));
-    };
-    input.click();
-  }
+  const handleCapture = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+    const files = target.files;
+    if (files) {
+      uploadImage({ file: files[0] })
+        .then(() => {
+          fetchProfileAndUpdateContext();
+          updateSnackBarMessage('Your photo is uploaded successfully');
+        })
+        .catch(() => ({
+          error: { message: 'Unable to connect to server. Please try again' },
+        }));
+    }
+  };
 
-  function removeFile() {
-    setFile('');
+  async function removeFile() {
+    const profile = loggedInUserProfile;
```

Comment:
```suggestion
    const profile = {...loggedInUserProfile}
```
Please avoid mutating the state directly in React (the variable 'profile' references the loggedInUserProfile object)

Reply:
I thought the subject in context will not be mutated. Haha. Anyway, using {...   } is definitely a good practice. Done.